Roommate-Budget-Helper/budget-helper-webui
# Rbh home crud design

## Jira Ticket: RBH-46

## Description of Work
Created a Homes Page, Create a Home Page, and View a Home Page for a logged in User.
## Acceptance Criteria

- Logged in User can create a Home
- Logged in User can see all Homes created on the Homes Page
- Logged in User can select any Home's "View" button and View the Home's data on the Home Page

## How to Test

- Log into a User and attempt to load "/homes" webpage
- When on "/homes" webpage, click the "+" sign on the screen to create a home
- Enter Home information on the Create a Home page and click "create"
- Ensure the "/homes" page you get returned to shows the newly created home.
- Click "view" on the Home page's data on "/homes"
- Ensure the Home page of the selected home and all respective data appears.

### List of Changes:

- Added database Home and Occupies object to track all Homes and who Occupies homes
- Fixed issues from creating Homes that did not attach new Home's to the correct User due to Cognito id issues
- Split Homes functionality between three files "homes.tsx", "home/[id].tsx", and "createhome.tsx"

### Screenshots

| Image | Description |
| 
![image](https://user-images.githubusercontent.com/73263545/214488739-8bb92e03-1ef6-454e-9659-cd4274470cd0.png)
 | User is logged in |
| 
![image](https://user-images.githubusercontent.com/73263545/214488832-46b2639b-72b0-4e36-8d83-f4fd6f572aff.png)
 | On "/homes" page |
| 
![image](https://user-images.githubusercontent.com/73263545/214488935-e0904a90-0b38-4564-b834-e8c26207b283.png)
 | Clicked the "+" button to Create a Home |
| 
![image](https://user-images.githubusercontent.com/73263545/214489003-d72d7ea6-4d26-4f25-a525-542ae7651c53.png)
 | Filled in Home Information |
| 
![image](https://user-images.githubusercontent.com/73263545/214489059-c8b9eee1-bd5e-426c-9d5f-cec39f935b1e.png)
 | Home Is added to "/homes" page |
| 
![image](https://user-images.githubusercontent.com/73263545/214489104-7395d97a-3bda-467c-b4ae-c6edfef928ad.png)
 | Clicked on "view" on the Home I wanted to view |
